Automattic/jetpack
# VideoPress: first approach of data handling

This PR implements the first approach to the data handling of the VideoPress package. Based on components/data (Redux).

Fixes #

#### Changes proposed in this Pull Request:

* VideoPress: first approach of data handling

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to VideoPress admin page
* Install Redux Dev Console: [Firefox](https://addons.mozilla.org/en/firefox/addon/reduxdevtools/) | [Chrome](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en)
* Open dev console
* Set debug scope:
```
localStorage.setItem( 'debug', 'jetpack-videopress:state' );
```
* Ensure you're able to show logs

<img width="184" alt="image" src="https://user-images.githubusercontent.com/77539/188687102-a637475f-32af-4a0c-a69e-a6ca3d6f9ab6.png">

* Confirm the redux state is initialized 

<img width="509" alt="image" src="https://user-images.githubusercontent.com/77539/188687357-f45f4258-91dc-49cf-8137-43111a4dd752.png">

* get videos by using the selector

```
wp.data.select( 'jetpack-videopress' ).getVideos()
```

* You should see undefined the first time. Do it again.
* Confirm you see the videos list.


https://user-images.githubusercontent.com/77539/188686187-b46aa8e3-69a7-4580-8c2e-6834709adcbc.mov
